### PR TITLE
disco: use offsets in txncache

### PR DESCRIPTION
**Pull request by arjain4**

### CODE REVIEW

File: src/disco/bank/fd_txncache.c [hunk 1]
```diff
@@ -324,61 +332,17 @@ fd_txncache_join( void * shtc ) {
...
-  }
-
-  fd_txncache_t * tc = (fd_txncache_t *)shtc;
-
-  if( FD_UNLIKELY( tc->magic!=FD_TXNCACHE_MAGIC ) ) {
-    FD_LOG_WARNING(( "bad magic" ));
-    return NULL;
-  }
+  uchar * base = (uchar *)tc;
+  tc->root_slots = (ulong *)( base + tc->root_slots_off );
```
Comment (mmcgee-jump) [inline src/disco/bank/fd_txncache.c:336]: ?? This isn't right, the different joins share this memory still. You need to remove these pointers from the struct and just reference everything from the offset always (probably with helper fns, see eg, the pattern used by fd_cnc_app_laddr

https://github.com/firedancer-io/firedancer/blob/main/src/tango/cnc/fd_cnc.h#L211

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
.

File: src/disco/bank/fd_txncache.c (hunk 1 above)
Comment (arjain4) [reply src/disco/bank/fd_txncache.c:336]: this change is made, testing against mainnet snapshot

**Review by lheeger-jump — APPROVED** [review]
I think this looks good but @mmcgee-jump might have more feedback

**Pull request merged**